src/dir.rs

```diff
@@ -468,10 +468,76 @@ impl Drop for TempDir {
     }
 }
 
-pub(crate) fn create(path: PathBuf) -> io::Result<TempDir> {
-    fs::create_dir(&path)
-        .with_err_path(|| &path)
-        .map(|_| TempDir {
-            path: path.into_boxed_path(),
-        })
+pub(crate) fn create(
+    path: PathBuf,
+    permissions: Option<&std::fs::Permissions>,
+) -> io::Result<TempDir> {
+    imp::create(path, permissions)
+}
+
+mod imp {
+    #[cfg(unix)]
+    mod unix {
+        use crate::TempDir;
+        use rustix::fs::Mode;
+        use std::io;
+        use std::os::unix::fs::PermissionsExt;
+        use std::path::PathBuf;
+
+        pub fn create(
+            path: PathBuf,
+            permissions: Option<&std::fs::Permissions>,
+        ) -> io::Result<TempDir> {
+            let mode: rustix::fs::Mode = permissions.map_or(
+                Mode::RWXU | Mode::RGRP | Mode::XGRP | Mode::ROTH | Mode::XOTH,
```

Comment:
I believe the correct thing to pass here is 0777, and let the umask deal with it (we're being inconsistent with tempfile.... but changing that now isn't a great idea).

Reply:
With `std::fs::DirBuilder` we have the luxury of not specifying it unless it's overridden, I will do that.